api/src/main/java/io/strimzi/api/kafka/model/balancing/CruiseControlBrokerCapacity.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.balancing;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Maximum;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Representation of the Cruise Control broker capacity settings. Since the Kafka brokers
+ * in Strimzi are homogeneous, the capacity values for each resource will be
+ * used for every broker.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"disk", "cpu", "networkIn", "networkOut"})
+@EqualsAndHashCode
+public class CruiseControlBrokerCapacity implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private Integer disk;
+    private Integer cpu;
+    private Integer networkIn;
+    private Integer networkOut;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    @Description("Broker capacity for disk in megabytes.")
+    public Integer getDisk() {
+        return disk;
+    }
+
+    public void setDisk(Integer disk) {
+        this.disk = disk;
+    }
+
+    @Minimum(0)
+    @Maximum(100)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("Broker capacity for CPU utilization as a percentage (0 - 100).")
+    public Integer getCpu() {
```

Comment:
Cruise Control treats the resources specified in requests and limits for each broker as an individual CPU resource. The CC CPU capacity value here is what percentage of a broker's CPU resource can be used without violating the broker's CPU capacity, whether that CPU resource is made up of one or more CPU cores.

Reply:
I was wrong about the method[1] respecting cgroup boundaries. It does not and it causes the CPU utilization of the brokers to be underestimated. I am currently looking into a way around this issue but in the meantime, it may be best for stripping this `cpuUtilization` configuration out of here before committing to a specific format. I may need to use this field to get around this problem.

Would it be alright if I removed this cpuUtilization field for this PR and add it in at a later point in time? Or do all the properties of `cruiseControl.brokerCapacity` have to all be added at the same time?


[1] https://docs.oracle.com/javase/7/docs/jre/api/management/extension/com/sun/management/OperatingSystemMXBean.html#getProcessCpuLoad()